modcc/printer/cudaprinter.cpp

```diff
@@ -445,7 +449,7 @@ void emit_state_update_cu(std::ostream& out, Symbol* from,
         out << "arb::gpu::reduce_by_key(";
         is_minus && out << "-";
         out << from->name()
-            << ", params_." << d.data_var << ", " << index_i_name(d.index_var) << ");\n";
+            << ", params_." << d.data_var << ", " << index_i_name(d.index_var) << ", lane_mask_);\n";
```

Comment:
Yes, it is. Before doing the full conversion to calculating run-length outside `reduce_by_key`, more tests need to be added to the cuda printer to ensure that it is only called on `APIMethod`s that perform reductions.

I was not comfortable using the `__activemask` intrinsic to find the mask inside `reduce_by_key`, because the docs didn't convince me that there wouldn't be divergent threads, even directly after `__syncwarp`.